python/pyspark/ml/param/__init__.py

```diff
@@ -280,7 +281,9 @@ def explainParam(self, param):
         else:
             values.append("undefined")
         valueStr = "(" + ", ".join(values) + ")"
-        return "%s: %s %s" % (param.name, param.doc, valueStr)
+        # Remove the default value from the doc string
+        docStr = re.sub(r'\(Default [\w\-\s]+?\)\Z', '', param.doc)
+        return "%s: %s %s" % (param.name, docStr, valueStr)
```

Comment:
So this filters the auto generated ones that get added. For other strings I think its up to the creator of the param to follow the format if they want the standard behaviour. I don't think we need to catch all of them this is just for nicer display during explain.